src/reitit/core.cljc

```diff
@@ -55,18 +55,25 @@
       (meta-merge acc {k v}))
     {} x))
 
-(defn resolve-routes [data {:keys [coerce] :or {coerce identity} :as opts}]
-  (->> (walk data opts)
-       (map-meta merge-meta)
-       (mapv (partial coerce))
-       (filterv identity)))
+(defn resolve-routes [data {:keys [coerce] :as opts}]
+  (cond-> (->> (walk data opts)
+               (map-meta merge-meta))
+          coerce (->> (mapv (partial coerce))
```

Comment:
Could be simpler and maybe even more performant with `(cond->> ... coerce (into [] (keep (partial coerce))))`.